language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/dependencygraph/DependencyGraph.scala

```diff
@@ -49,10 +55,20 @@ private[codegen] object DependencyGraph {
     * targeting template definitions that can be observed through
     * the Ledger API.
     */
-  def transitiveClosure(decls: Map[Identifier, InterfaceType]): TransitiveClosure = {
-    val dependencies = orderedDependencies(decls)
+  def transitiveClosure(
+      serializableTypes: Map[Identifier, InterfaceType],
+      interfaces: Map[Identifier, DefInterface.FWT],
+  ): TransitiveClosure = {
+    val dependencies = orderedDependencies(serializableTypes, interfaces)
     TransitiveClosure(
-      interfaceTypes = dependencies.deps.map { case (id, Node(t, _, _)) => id -> t },
+      orderedDependencies = orderedDependencies(serializableTypes, interfaces).deps.map {
```

Comment:
Duplicated logic that needs to be fixed.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/13557/commits/9a42878f62ee9cafb6f61e7504300807a5cc9c54